### PR TITLE
Add Find function #12

### PR DESCRIPTION
**Pull request by flyznex**
## Checklist for PR
 #12
[x] Link to corresponding Issue.
[x] Tests covering all the code branches.
[x] Validations for all the argument's type. This includes the functions passed as arguments.
[x] A Comment on how the function works and what are the different validations in the function.
[x] An ExampleTest of the function.
[x] README.md should be updated about the function.

### CODE REVIEW

File: find_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"github.com/thecasualcoder/godash"
+)
+
+func TestFind(t *testing.T) {
+	t.Run("should filter elements that fail predicate", func(t *testing.T) {
+		input := []int{1, 2, 3, 4, 5, 6, 7, 8}
+		var output int
+
+		err := godash.Find(input, &output, func(a int) bool {
+			return a == 1 // starts with
```
Comment (dineshba) [inline find_test.go:18]: remove comment //starts with

File: find.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+			returnValues := predicate.Call([]reflect.Value{arg})
+			predicatePassed := returnValues[0].Bool()
+
+			if predicatePassed {
+				output.Elem().Set(arg)
+				return nil
+			}
+		}
+
```
Comment (dineshba) [inline find.go]: we have to `return nil` here. Could you write a test covering this case?(basically element not found in the given list)

File: find.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import (
+	"fmt"
+	"reflect"
+)
+
+// Find out elements.
+//
+// Input of type slice is supported as of now.
+// Output is a elements are matched.
+// PredicateFn function is applied on each element of input to determine to find element
```
Comment (dineshba) [inline find.go]: // PredicateFn function is applied on each element of input to determine to find element until it finds the element

**Review by dineshba — CHANGES_REQUESTED** [review]

File: find.go [hunk 4]
```diff
@@ -55,7 +55,7 @@ func Find(in, out, predicateFn interface{}) error {
 				return nil
 			}
 		}
-
+		return fmt.Errorf("element not found")
```
Comment (dineshba) [inline find.go:58]: @aswinkarthik Should we return error if the element is not found?

**Review by dineshba — APPROVED** [review]

File: find.go (hunk 4 above)
Comment (aswinkarthik) [reply find.go:58]: Yes. In order to differentiate can we have a separate ErrElementNotFound so that it can be distinguished from validation errors?

Comment (dineshba) [reply find.go:58]: Okay. I agree. @flyznex Could we do it as part of this PR? What do you think?

Comment (flyznex) [reply find.go:58]: @dineshba I think we need create new file for define, format errors. In this PR, ours issue has been solved. We will create separate Issue, PR for error handling.

Comment (dineshba) [reply find.go:58]: Sure @flyznex

**Pull request merged**